walletnode/services/artworkregister/task.go

```diff
@@ -149,7 +151,11 @@ func (task *Task) meshNodes(ctx context.Context, nodes node.List, primaryIndex i
 					if err := node.Session(ctx, false); err != nil {
 						return
 					}
+
+					//data race detection without lock
+					mtx.Lock()
```

Comment:
### Not Cheating

You cannot inject code directly into tested func only to pass unit test. 
You can change the code only if you find a bug. this is definitely not a bug.

Reply:
Ok got it. Going to change it. I am not intent to cheating. I thought it was a bug.

Reply:
Ok. just to make sure this is not a bug, show me the error, and I will tell for sure

Reply:
this is what i saw when CI check running https://app.circleci.com/pipelines/github/pastelnetwork/gonode/460/workflows/987aa2bb-57cb-4f33-b1bf-65b69da79aae/jobs/1214 few days ago 

but right now everything is fine without add mux.lock before adding node ( after I updated the client mock workflow). I think i was not create proper mock workflow for meshNodes method.

Reply:
BTW I added t.Skip() at two places, did you not forget to remove it?

Reply:
ah my bad. i only noticed skip on TestTaskRun method. I did not noticed on TestTaskMeshNodes. going to try removed it and  watching circlecitest

Reply:
> this is what i saw when CI check running https://app.circleci.com/pipelines/github/pastelnetwork/gonode/460/workflows/987aa2bb-57cb-4f33-b1bf-65b69da79aae/jobs/1214 few days ago
> 
> but right now everything is fine without add mux.lock before adding node ( after I updated the client mock workflow). I think i was not create proper mock workflow for meshNodes method.

It looks like the same mock is used in parallel test (race), I assume  you need to make sure, a new mock is created for each testCase